### PR TITLE
feat!: Migrate GitHub plugin to v2

### PR DESCRIPTION
**Pull request by candiduslynx**
#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Test locally on your own infrastructure
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation)) 
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: plugins/source/github/docs/index.md [hunk 1]
```diff
@@ -19,16 +16,12 @@ CloudQuery needs to be authenticated with your GitHub account's Personal Token i
...
-    access_token: "<YOUR ACCESS TOKEN HERE>"
-    orgs: ["<YOUR ORG NAME>"]
-  resources:
-    - "*"
+First, generate the configuration template using the following command:
+```bash
+cloudquery gen source github
 ```
 
+After that up fill in the `access_token` and `orgs` values.
```
Comment (hermanschaaf) [inline plugins/source/github/docs/index.md:24]:
```suggestion
After that, edit `github.yml` and set the `access_token` and `orgs` values. 
```

File: plugins/source/github/resources/plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cloudquery/cloudquery/plugins/source/github/client"
+	"github.com/cloudquery/plugin-sdk/plugins"
+)
+
+const (
+	exampleConfig = `
+		access_token: <Access Token HERE>
+		orgs:
+		- cloudquery
+`
```
Comment (hermanschaaf) [inline plugins/source/github/resources/plugin/plugin.go:13]:
```suggestion
	exampleConfig = `
access_token: <Access Token HERE>
orgs:
- cloudquery
`
```

I believe this should not be indented

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
I didn't review all the resources in great detail (I think we can merge and fix any issues as they come up). Overall it looks good :+1: I just left a couple of comments

File: plugins/source/github/resources/plugin/plugin.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/github/resources/plugin/plugin.go:13]: When we use indentation then we get the properly constructed example config.
Alternatively, we should go to filling in the Spec with the example data and marshal it to YAML altogether.

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Given we have some more time for v2 release, let's introduce the new codegen. We only generate the structs and the resolvers should go into `resource_fetch.go`. See example - https://github.com/cloudquery/cloudquery/pull/1774

File: plugins/source/github/resources/plugin/plugin.go (hunk 2 above)
Comment (hermanschaaf) [reply plugins/source/github/resources/plugin/plugin.go:13]: @candiduslynx I'm not sure how you checked, but using this branch, I just ran the plugin server with `go run main.go serve` and then use the cli to run `go run main.go generate source --registry grpc 127.0.0.1:7777`. The generated config was not correctly indented:

Screenshot 2022-09-12 at 08 55 36

The code for the `generate` command adds the necessary indentation to get it at the right indentation level, we don't need to do it here. 

Personally I think an example string is fine, but if you want to change it to struct marshalling to yaml, that's great too, let's just make sure there's no indent either way.

File: plugins/source/github/resources/plugin/plugin.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package plugin
+
+import (
+	"github.com/cloudquery/cloudquery/plugins/source/github/client"
+	"github.com/cloudquery/plugin-sdk/plugins"
+)
+
+const (
+	exampleConfig = `
+		access_token: <Access Token HERE>
```
Comment (yevgenypats) [inline plugins/source/github/resources/plugin/plugin.go:10]: This should be with no tab - otherwise `gen` wont work.

**Review by yevgenypats — APPROVED** [review]
Looks good, I might have some more comments and nits on gen improvements but I think we can do this later on.

File: plugins/source/github/resources/plugin/plugin.go (hunk 3 above)
Comment (yevgenypats) [reply plugins/source/github/resources/plugin/plugin.go:10]: cc @candiduslynx

Comment (candiduslynx) [reply plugins/source/github/resources/plugin/plugin.go]: done

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**